File: 4. Java/EX48.java
package Apostila;

import java.util.Locale;
import java.util.Scanner;

public class EX48
{
    public static void main(String[] args)
    {
        // Variáveis
        int qtd, qtdAbaiMed = 0;
        double acuSom = 0, med;
        
        // Define o locale para os EUA
        Locale.setDefault(Locale.US);
        
        // Lê a quantidade de elementos do vetor
        try (Scanner ent = new Scanner(System.in))
        {
            System.out.print("# Quantos elementos vai ter o vetor: ");
            qtd = ent.nextInt();
        
            double[] vetNum = new double[qtd], aux = new double[qtd];
            
            // Lê os números e acumula a soma
            for (int i = 0; i < qtd; i++)
            {
                System.out.print("# Digite um numero: ");
                vetNum[i] = ent.nextDouble();
                acuSom += vetNum[i];
            }
            
            // Calcula a média
            med = acuSom / qtd;
            
            // Identifica elementos abaixo da média
            for (int i = 0; i < qtd; i++)
            {
                if (vetNum[i] < med)
                {
                    aux[qtdAbaiMed] = vetNum[i];
                    qtdAbaiMed++;
                }
            }
            
            // Exibe a média do vetor
            System.out.printf("\n# MEDIA DO VETOR = %.3f", med);
            
            // Exibe elementos abaixo da média
            System.out.print("\n# ELEMENTOS ABAIXO DA MEDIA:");
            for (int i = 0; i < qtdAbaiMed; i++)
            {
                System.out.printf("\n%.1f", aux[i]);
            }
            System.out.println("");
        }
    }
}

File: 6. Python/EX48.py
# Inicializa variáveis de controle
idAux = 0
acuSom = 0
qtdAbaiMed = 0

# Solicita a quantidade de elementos do vetor
qtd = int(input("# Quantos elementos vai ter o vetor: "))

# Cria vetores para os números e elementos abaixo da média
aux = [0.0 for i in range(qtd)]
vetNum = [0.0 for i in range(qtd)]

# Coleta os números e calcula a soma
for i in range(qtd):
    vetNum[i] = float(input("# Digite um numero: "))    # Coleta o número
    acuSom += vetNum[i]                                 # Acumula a soma

# Calcula a média
med = acuSom / qtd

# Encontra elementos abaixo da média
for i in range(qtd):
    if vetNum[i] < med:
        aux[idAux] = vetNum[i]  # Armazena o elemento abaixo da média
        idAux += 1              # Incrementa o índice
        qtdAbaiMed += 1         # Conta quantos estão abaixo da média

# Exibe a média do vetor
print(f"\n# MEDIA DO VETOR = {med:.3f}")

# Exibe os elementos abaixo da média
print("# ELEMENTOS ABAIXO DA MEDIA:")

for i in range(qtdAbaiMed):
    print(f"{aux[i]:.1f}")  # Exibe elemento abaixo da média
